### ISSUE TITLE
The JAX traceability probe traces a bare call, not the vmap dispatch actually runs

### ISSUE THREAD

**Issue by jhuggins** [TARPS-group/prob-pipe#403]
## What happens

`Function`'s automatic dispatch decides between `jax` and `sequential` by probing
whether the body is JAX-traceable. The probe traces a **bare call**:

```python
jax.make_jaxpr(lambda kw: func(**kw))(dummy_kw)
```

but what it is deciding about is a **`jax.vmap`** over that body. A body that
traces fine unbatched but is incompatible with `vmap` therefore passes the probe,
`jax` is chosen, and the call fails at execution instead of falling back.

## A body that hits it

A function whose body returns a `RecordBatch`:

```python
@function
def f(v):
    return NumericRecordBatch(
        {"s": jnp.asarray(v["x"]) + jnp.zeros(2)},
        "inner",
        element_spec=EventTemplate(s=()),
        axis_groups=((2,),),
    )

f(v=some_batch)   # ValueError from unflattening
```

`vmap` adds an axis to each column on the way out. That axis belongs to no level,
and unflattening has no name to give one, so it raises. The sequential and thread
paths handle the same call correctly, returning a batch over
`("<the swept level>", "inner")`.

## Why the fallback does not catch it

`Function._resolve_dispatch` already falls back on a probe error:

```python
if self._jax_traceability_error(values, broadcast_args, func=func) is None:
    return "jax"
logger.info("Function '%s' is not JAX-traceable; using sequential dispatch.", self._name)
return "sequential"
```

The machinery is right; the probe is just not representative of what runs.

## Proposal

Trace the probe under `vmap` over a stacked dummy row, so it exercises the
transform the dispatch is choosing. The existing probe-to-sequential fallback then
covers this case and any other `vmap`-only incompatibility, with the INFO log it
already emits.

Worth reviewing on its own rather than inside a value-type change: the probe
governs every automatic dispatch decision, not only ones involving batches, and
tracing a `vmap` costs more than tracing a bare call.

## Scope

- `probpipe/core/node.py` — `Function._jax_traceability_error`
- Tests: a body that traces bare but not under `vmap` should resolve to
  `sequential` rather than raise; bodies that vmap fine should still resolve to
  `jax`, so the change does not silently move everything to the slow path.
